src/Pos/Genesis.hs

```diff
@@ -109,35 +109,73 @@ stakeDistribution ts@RichPoorStakes {..} =
         else identity
     basicDist = genericReplicate sdRichmen sdRichStake ++
                 genericReplicate sdPoor sdPoorStake
-stakeDistribution (CustomStakes coins) = map (,[]) coins
-
--- | Generates genesis 'Utxo' given optional boot stakeholders list
--- and address distribution. In case boot stakeholders are supplied,
--- all the balance is distributed among them. Otherwise, txDistr is
--- set to @[]@ or left as it is (in case of 'ExplicitStakes').
-genesisUtxo :: Maybe (HashMap StakeholderId Word16) -> [AddrDistribution] -> Utxo
-genesisUtxo bootStakeholders ad =
-    M.fromList $ concatMap (uncurry toUtxo . second stakeDistribution) ad
+stakeDistribution (CustomStakes coins) = coins
+
+-- | Generates genesis 'Utxo' given optional boot stakeholders (with
+-- weights) and address distributions.  If genesis stakeholders are
+-- not supplied, they are calculated from address distributions by
+-- making each 'PubKeyAddress' genesis stakeholder with weight equal
+-- to balance. All the stake is distributed among genesis stakeholders
+-- (using 'genesisSplitBoot').
+--
+-- TODO [CSL-1351] This documentation will become correct later.
+genesisUtxo ::
+       Maybe (HashMap StakeholderId Word16) -> [AddrDistribution] -> GenesisUtxo
+genesisUtxo bootStakeholdersMaybe ad
+    -- TODO [CSL-1351] Uncomment ↓.
+    --- | null bootStakeholders =
+    ---     error "genesisUtxo: no stakeholders for the bootstrap era"
+    -- TODO [CSL-1351] Delete (it's here to please hlint) ↓.
+    | ((*) @Int) 2 3 == 7 = error "hlint — makaka"
```

Comment:
There error is useless guard. It should be removed when commented code it uncommented (and `| null bootStakeholders` should take place.